channels/main.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+)
+
+var numbers = []string{"1", "2", "3", "4", "5", "6", "7", "8", "9"}
+
+func main() {
+	strings := []string{"1211Nikhil34Shar2ma", "H33a4r6shi777tS7687harma", "S45346eem568568aSh86a9r8ma"}
+	c := make(chan string)
+	correctStrings := getCorrectStrings(strings, c)
+	fmt.Println(correctStrings)
+}
+
+func getCorrectStrings(inputStrings []string, c chan string) []string {
+	var filteredString string
+	var filteredStrings []string
+
+	for _, s := range inputStrings {
+		go func(s string) {
+			filteredString = numericFreeString(s, c)
+			filteredStrings = append(filteredStrings, filteredString)
+			c <- "ok"
+		}(s)
+	}
+	for i := 0; i < len(inputStrings); i++ {
+		<-c
+	}
```

Comment:
This will work but it can be optimized. The for loop at line 28 is just there to prevent the deadlock, otherwise not required. Try thinking in terms of buffered channels to fix deadlock situation.

Reply:
We can add buffered channel waiting for the message in the for-loop at line 21 only, after the 
```
go func(s string) {
			filteredString = numericFreeString(s, c)
			filteredStrings = append(filteredStrings, filteredString)
			c <- "ok"
}(s)
```
But that will not allow concurrency in the logic.